common/src/main/java/org/wso2/testgrid/common/DeploymentPattern.java

```diff
@@ -79,6 +88,9 @@
     @OneToMany(mappedBy = "deploymentPattern", cascade = CascadeType.ALL, orphanRemoval = true)
     private List<TestPlan> testPlans = new ArrayList<>();
 
+    @OneToMany(mappedBy = "deploymentPattern", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<DeploymentPatternResourceUsage> deploymentPatternResourceUsageList = new ArrayList<>();
```

Comment:
A deployment pattern is a generic concept. Tying that with a resource 'usage' is not correct. We have to rethink how you have integrated your implementation into the testgrid core architecture.